driver-core/src/test/functional/com/mongodb/internal/binding/OperationContextBinding.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.binding;
+
+import com.mongodb.ReadPreference;
+import com.mongodb.connection.ServerDescription;
+import com.mongodb.internal.connection.Connection;
+import com.mongodb.internal.connection.OperationContext;
+
+import static org.bson.assertions.Assertions.notNull;
+
+public class OperationContextBinding implements ReadWriteBinding {
+    private final ReadWriteBinding wrapped;
+    private final OperationContext operationContext;
+
+    public OperationContextBinding(final ReadWriteBinding wrapped, final OperationContext operationContext) {
+        this.wrapped = notNull("wrapped", wrapped);
+        this.operationContext = operationContext;
+    }
+
+    @Override
+    public ReadPreference getReadPreference() {
+        return wrapped.getReadPreference();
+    }
+
+    @Override
+    public int getCount() {
+        return wrapped.getCount();
+    }
+
+    @Override
+    public ReadWriteBinding retain() {
+        wrapped.retain();
```

Comment:
Minor, but is it worth having a test for this class (the retain/release behaviour)?